test/perf/perf_s3_client.cc

```diff
@@ -80,35 +86,46 @@ class tester {
         }
     }
 
-private:
+public:
+    future<> start() {
+        if (_object_name.empty()) {
+            co_await make_temporary_file();
+        }
+    }
 
-    future<> do_run() {
+    future<> run_download() {
+        plog.info("Downloading");
         auto until = now() + _duration;
         uint64_t off = 0;
         do {
             auto start = now();
             try {
                 co_await _client->get_object_contiguous(_object_name, s3::range{off, chunk_size});
                 off = (off + chunk_size) % (_object_size - chunk_size);
-                _reads_hist.add(std::chrono::duration_cast<std::chrono::milliseconds>(now() - start).count());
+                _latencies.add(std::chrono::duration_cast<std::chrono::milliseconds>(now() - start).count());
```

Comment:
Ran this test multiple times, but I always get zeroes in the stats print. Did you manage to get any stats?

Reply:
For upload _latencies are not updated. Need more work here, namely -- pull latencies all the way down to s3::client::multipart_upload::<methods> and make them update the histogram